### DISCUSSION TITLE
About using mg_mgr_poll cause FD_SETSIZE overflow

### DISCUSSION THREAD

**Discussion by BaronStack** [cesanta/mongoose#1457] [Ask questions, get answers] [answered]
os : centos 7.8
purpose: mongoose mg_mgr_poll for http_server

Our http server process implemented with `mg_mgr_poll` function.
We can see the mg_mgr_poll function is using  **select** system call to deal the new connection.
But our process have more than 1024 files when running, so the select FD_SET is overflow.

Is there any reason for you to use the **select** system call rather than  poll / epoll?
Do you have the plan to support it ?

**Comment by cpq** [comment, marked as answer]
There are no plans to use poll/epoll.
Search how to increase FD_SETSIZE in your environment.

**Reply by vkoskiv** [reply]
Hey! I started using mongoose for a [hobby project](https://github.com/vkoskiv/nmc2) a while back, and it's been working great so far. I'm a bit bummed about this `FD_SETSIZE` limit that I didn't anticipate when I started. I've considered making a downstream fork that uses `poll()`, but before I start investigating that, I'd love to hear more about what drove this decision on your end - some technical limitation or just a desire to keep the code simple?

I saw #439 from a long time ago where using `poll()` seemed to be on the table, and PR #508 introduced it, but I couldn't find the discussion/rationale for removing it later.
In any case, thanks for this project! Excellent documentation and really nice API design.

**Reply by vkoskiv** [reply]
@cpq just in case my reply didn't ping you.
